### DISCUSSION TITLE
Interaction with this project and the IntelliJ profiler

### DISCUSSION THREAD

**Discussion by CLOVIS-AI** [Anamorphosee/stacktrace-decoroutinator#41] [General]
The IntelliJ profiler uses `async-profiler` by default, which works on stacktraces. That makes it particularly unhelpful in coroutines projects, as stacktraces don't mean anything.

Does the use of stacktrace-decoroutinator make the profiler usable again?

**Comment by Anamorphosee** [comment]
Sure. The IntelliJ profiler also displays the full stack trace when Decoroutinator is enabled.

**Reply by CLOVIS-AI** [reply]
I meant, is the profiler able to generate a meaningful flamegraph with decoroutinator?

**Reply by Anamorphosee** [reply]
Seems I got you. No, Frame Graph is still the same as without Decoroutinator. Because coroutines' logic doesn't change, suspending and resuming of a coroutine still requires ending and starting its methods. The only thing Decoroutinator changes is resuming: when a coroutines resumes it calls its full trace instead of its top only.

**Reply by CLOVIS-AI** [reply]
That's a shame. It would be really nice to have a way to profile coroutines…
